### PR TITLE
Adding more nifgen systemtests

### LINKED ISSUE #484: Write system tests for NI-FGEN

### PR DESCRIPTION
**Pull request by injaleea**
[X] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/nimi-python/blob/master/CONTRIBUTING.md).

~~[ ] I've updated [CHANGELOG.md](https://github.com/ni/nimi-python/blob/master/CHANGELOG.md) if applicable.~~

[X] I've added tests applicable for this pull request

### What does this Pull Request accomplish?

Adding 8 more system tests for nifgen

- test_allocate_waveform
- test_clear_waveform_memory
- test_query_arb_seq_capabilities
- test_arb_seq
- test_arb_script
- test_reset
- test_reset_device
- test_reset_with_default

### List issues fixed by this Pull Request below, if any.
* Partial work towards #484 

### What testing has been done?

systemtests, tox

### CODE REVIEW

**Comment by marcoskirsch** [comment]
Changed title to "Adding more nifgen systemtests"

* There's a driver called NI-FGEN
* There's a Python module called nifgen
* There's nothing called fgen. We should use unambiguous terms.

Under "List issues fixed by this Pull Request below, if any." you say it fixes #484. Does it **completely** fix it? Because if it does not, then you need to change the wording of that bullet else GitHub will close the issue.

File: src/nifgen/system_tests/test_system_nifgen.py [hunk 1]
```diff
@@ -148,3 +148,22 @@ def test_query_freq_list_capabilities(session):
 
 def test_read_current_temperature(session):
     assert session.read_current_temperature() > 25.0
+
+
+def test_allocate_waveform(session):
+    session.allocate_waveform(1024)
```
Comment (marcoskirsch) [inline src/nifgen/system_tests/test_system_nifgen.py]: Verify this returns a valid waveform handle. You may need to do something to it afterwards, maybe an assert.

File: src/nifgen/system_tests/test_system_nifgen.py [hunk 2]
```diff
@@ -148,3 +148,22 @@ def test_query_freq_list_capabilities(session):
...
+def test_clear_waveform_memory(session):
+    session.clear_arb_memory()
+    session.clear_user_standard_waveform()
+    session.clear_arb_sequence(-1)
+    session.clear_arb_waveform(-1)
+
+
+def test_query_arb_seq_capabilities(session):
+    maximum_number_of_sequences, minimum_sequence_length, maximum_sequence_length, maximum_loop_count = session.query_arb_seq_capabilities()
+    assert maximum_number_of_sequences == 65535  # maximum_number_of_sequences, minimum_sequence_length, maximum_sequence_length, maximum_loop_count are 65535, 1, 65535, 16777215 respectively for simulated 5433
```
Comment (marcoskirsch) [inline src/nifgen/system_tests/test_system_nifgen.py]: I think the comment is just paraphrasing the code which is very readable on its own, you can remove.

**Comment by marcoskirsch** [comment]
> Under "List issues fixed by this Pull Request below, if any." you say it fixes #484. Does it completely fix it? Because if it does not, then you need to change the wording of that bullet else GitHub will close the issue.

I looked at nimi-bot's output. Coverage is at 65% for nifgen so this doesn't yet fully "fix" the issue. I've changed the wording on the PR description.

```
 generated xml file: C:\Jenkins\workspace\nimi_python_system_tests\bin\junit\junit-modinst-system_tests.xml 
========================== 15 passed in 1.48 seconds ==========================
Coverage.py warning: --include is ignored because --source is set (include-ignored)
system_tests runtests: commands[12] | coverage report --rcfile=tools/coverage_system_tests.rc
Name                                                       Stmts   Miss  Cover
------------------------------------------------------------------------------
.tox\system_tests\Lib\site-packages\nidcpower\library.py     399     18    95%
.tox\system_tests\Lib\site-packages\nidmm\library.py         407     12    97%
.tox\system_tests\Lib\site-packages\nifgen\library.py        583    204    65%
.tox\system_tests\Lib\site-packages\nimodinst\library.py      47      0   100%
.tox\system_tests\Lib\site-packages\niscope\library.py       647    468    28%
.tox\system_tests\Lib\site-packages\niswitch\library.py      327      0   100%
------------------------------------------------------------------------------
TOTAL                                                       2410    702    71%
system_tests runtests: commands[13] | coverage html --rcfile=tools/coverage_system_tests.rc --directory=bin/htmlcov/system_tests
___________________________________ summary ___________________________________
  system_tests: commands succeeded
  congratulations :)
```

File: src/nifgen/system_tests/test_system_nifgen.py (hunk 1 above)
Comment (injaleea) [reply src/nifgen/system_tests/test_system_nifgen.py]: Easier way to validate the changes will be writing more data than allocated.

**Pull request merged**